guides/html-tag-processor.md

```diff
@@ -0,0 +1,208 @@
+	accessibleText: {
+		type: 'string',
+		default: '',
+	}
+};
+
+/**
+ * BlockEdit
+ *
+ * a react component that will get mounted in the Editor when the block is
+ * selected. It is recommended to use Slots like `BlockControls` or `InspectorControls`
+ * in here to put settings into the blocks toolbar or sidebar.
+ *
+ * @param {object} props block props
+ * @returns {HTMLElement}
+ */
+const BlockEdit = (props) => {
+	const {
+		attributes: { accessibleText },
+		setAttributes,
+	} = props;
+
+	return (
+		<InspectorControls>
+			<PanelBody title={__('Additional Information')}>
+				<TextareaControl
+					value={accessibleText}
+					label={__('Accessible Text')}
+					help={__(
+						'This replaces what screenreader will read as the link text.',
+						'tenup-theme',
+					)}
+					onChange={(value) => {
+						setAttributes({
+							accessibleText: value,
+						});
+					}}
+				/>
+			</PanelBody>
+		</InspectorControls>
+	);
+};
+
+registerBlockExtension(blockName, {
+	extensionName: 'button-updates',
+	attributes: additionalAttributes,
+	classNameGenerator: () => null,
+	inlineStyleGenerator: () => null,
+	Edit: BlockEdit,
+});
+```
+
+```php title="block-overrides/core-button.php"
+<?php
+/**
+ * Core Button Block.
+ *
+ * @package TenUpTheme
+ */
+
+namespace TenUpTheme\BlockOverrides\CoreButton;
+
+/**
+ * Default setup routine
+ *
+ * @return void
+ */
+function setup(): void {
+	add_filter( 'render_block_core/button', __NAMESPACE__ . '\maybe_insert_aria_label', 10, 2 );
+}
+
+/**
+ * Add `aria-label` attribute to button.
+ *
+ * @param string $content Block content.
+ * @param array  $block   Block data.
+ *
+ * @return string
+ */
+function maybe_insert_aria_label( string $content, array $block ) : string {
+	$text = $block['attrs']['accessibleText'] ?? false;
+	$text = $text ? strval( $text ) : '';
+	if ( empty( $text ) ) {
+		return $content;
+	}
+
+	if ( false !== strpos( $content, 'aria-label' ) ) {
+		return $content;
+	}
+
+	$button = new \WP_HTML_Tag_Processor( $content );
+
+	// Perform a lookup for the anchor tag 
+	$query = array(
+		'tag_name' => 'A',
+	);
+
+	// No tag is selected until the $xyz->next_tag(); is called
+	$button->next_tag( $query );
+	$button->set_attribute( 'aria-label', $text );
+	$button->set_attribute( 'title', 'I am a title' );
+
+	return $button->get_updated_html();
+}
+
+/**
+ * Register accessibleText attribute to all blocks
+ * Assists with ServerSideRender
+ * REST API does data validation against the registered attributes
+ * Prevents throwing an error when unknown attributes get passed in
+ */
+function register_accessibletext_attribute_for_blocks() {
+	$registered_blocks = \WP_Block_Type_Registry::get_instance()->get_all_registered();
+
+	foreach ( $registered_blocks as $name => $block ) {
+		$block->attributes['accessibleText'] = array( 'type' => 'string' );
+	}
+}
+
+add_filter( 'wp_loaded', __NAMESPACE__ . '\\register_accessibletext_attribute_for_blocks', 999 );
```

Comment:
@fabiankaegy please check me on this function and the comment as I used the example you provided around `animation` as a guideline but I am not sure what I did is totally correct. Unlike the `animation` scenario in this use case we would just get a string back for the attribute of accessibleText

Reply:
@pdavies88 What you did for the attribute itself is correct :) 

The only thing I would change is that both in the commend and in the actual code you are adding the additional attribute to all blocks. In this case we only want to add it to one particular block. We should update that before we merge this :)

Reply:
@fabiankaegy I updated it to be wrapped in an if statement. Let me know if you think that is the correct approach:
```
/**
 * Register accessibleText attribute to core/button
 * Assists with ServerSideRender
 * REST API does data validation against the registered attributes
 * Prevents throwing an error when unknown attributes get passed in
 */
function register_accessibletext_attribute_for_blocks() {
	$registered_blocks = \WP_Block_Type_Registry::get_instance()->get_all_registered();
	foreach ( $registered_blocks as $name => $block ) {
		if ( 'core/button' === $block->name ) {
			$block->attributes['accessibleText'] = array( 'type' => 'string' );
		}
	}
}
add_filter( 'wp_loaded', __NAMESPACE__ . '\\register_accessibletext_attribute_for_blocks', 999 );
```